src/index.js

```diff
@@ -65,6 +69,69 @@ class BaseCommand extends Command {
     }
   }
 
+  logJson(message = '', ...args) {
+    /* Only run json logger when --json flag present */
+    if (!argv.json) {
+      return
+    }
+    process.stdout.write(JSON.stringify(message, null, 2))
+  }
+
+  log(message = '', ...args) {
+    /* If  --silent or --json flag passed disable logger */
+    if (argv.silent || argv.json) {
+      return
+    }
+    message = typeof message === 'string' ? message : inspect(message)
```

Comment:
We can use [`strip-ansi`](https://www.npmjs.com/package/strip-ansi) package to remove colors from the output based on some flag e.g. `--no-colors`. This will be helpful for situation where terminal colors are not supported like Travis and parsing output.

Reply:
Just tried this. it will work with normal log but not json logger/complex objects

It looks like the proxy is the way to go for now. In future, I'd love to use normal chalk options when we upgrade but it wasn't working with current versions